courses/serializers.py

```diff
@@ -370,12 +377,16 @@ def get_requirements(self, instance):
                     )
 
             else:
-                formatted_reqs["electives"] = [
-                    req.course.id
-                    for req in node.get_children()
-                    .filter(node_type=models.ProgramRequirementNodeType.COURSE)
-                    .all()
-                ]
+                formatted_reqs["electives"] = (
```

Comment:
This check chunk of code is similar to both electives and required. Could we refactor it to take it above the if and else?

Reply:
Pulled this all into `_get_nested_requirements` in https://github.com/mitodl/mitxonline/pull/1223/commits/2f44857a3c6db05c871e164098074d19bb680d19 (and added an additional check to make sure the node has a course ID)